ladybug/windrose.py

```diff
@@ -0,0 +1,953 @@
+        """
+        if self._label_point_size_factor is None:
+            self._label_point_size_factor = 0.8
+        return self._label_point_size_factor
+
+    @label_point_size_factor.setter
+    def label_point_size_factor(self, label_point_size_factor):
+        """Set the size of the points for labels in the windrose plot."""
+        self._label_point_size_factor = label_point_size_factor
+
+    @property
+    def label_tick_size_factor(self):
+        """Get the size of the ticks for labels in the windrose plot.
+
+        Default is 0.3
+        """
+        if self._label_tick_size_factor is None:
+            self._label_tick_size_factor = 0.3
+        return self._label_tick_size_factor
+
+    @label_tick_size_factor.setter
+    def label_tick_size_factor(self, label_tick_size_factor):
+        """Set the size of the ticks for labels in the windrose plot."""
+        self._label_tick_size_factor = label_tick_size_factor
+
+    @property
+    def legend_parameters(self):
+        """The LegendParameters object for this plot.
+
+        Default is a legend parameter with minimum and maximum values of the
+        windrose analysis values. If show_stack is False, then these min and max
+        will be the average of the binned values per orientation.
+        The default segment count is 10.
+        """
+        if self._legend_parameters is None:
+            if self.show_stack:
+                min_analysis = self.analysis_data_collection.min
+                max_analysis = self.analysis_data_collection.max
+            else:
+                # If not stacked histogram, ensure legend parameter range is defined
+                # by wind mean range per orientation.
+                wind_means = [sum(bin)/float(len(bin)) for bin in self.data
+                              if len(bin) > 0]
+                min_analysis = min(wind_means)
+                max_analysis = max(wind_means)
+            segment_count = 10
+            self._legend_parameters = LegendParameters(min=min_analysis,
+                                                       max=max_analysis,
+                                                       segment_count=segment_count)
+        return self._legend_parameters
+
+    @legend_parameters.setter
+    def legend_parameters(self, legend_parameters):
+        """Set the legend parameters for the windrose plot.
+
+        Args:
+            legend_parameters: A LegendParameters object.
+        """
+        legtyp = type(LegendParameters(0, 1, 10))
+        assert isinstance(legend_parameters, legtyp), 'legend_parameters must be' \
+            ' a LegendParameters. Got {}.'.format(type(legend_parameters))
+        self._legend_parameters = legend_parameters
+
+    @property
+    def legend(self):
+        """The Legend object for this plot"""
+        if self._legend is None:
+            self._legend = self._container._legend
+        return self._legend
+
+    @legend.setter
+    def legend(self, legend):
```

Comment:
Just to clarify, I think we can allow users to set the legend parameters, and any changes to that will propagate to the WindRose's Legend and GraphicContainer objects since Python passes pointers to complex data types. In theory we shouldn't have to manually regenerate the objects. It's a little confusing since we're counting on mutations to do all the work for us, but that's unavoidable either way.

Reply:
Nevermind. It looks like the radius is being used to intialize this object so my comment no longer applies (it's not just a object that sits within another object that chagnes, but the actual container class itself which will change.)

Reply:
Yes. Sorry I should have clarified this. You can see [the Legend object also duplicates any legend parameters that are passed to it](https://github.com/ladybug-tools/ladybug/blob/master/ladybug/legend.py#L116). So, this way, the original legend parameters are not mutated as defaults are set. In other words, if you want to edit the legend parameters assigned to a particular legend or graphic container, you have to get the legend_parameters on those objects themselves.